File: cpp/CPU.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;

class CPU : public Komponen
{
private:
    string nama_seri;
    int generasi;
    int jumlah_core;

public:
    CPU() {}

    CPU(string nama_seri, int generasi, int jumlah_core, string nama, string merk, int tahun_rilis, int harga) : Komponen(nama, merk, tahun_rilis, harga)
    {
        this->nama_seri = nama_seri;
        this->generasi = generasi;
        this->jumlah_core = jumlah_core;
    }

    void setNamaSeri(string nama_seri)
    {
        this->nama_seri = nama_seri;
    }

    void setGenerasi(int generasi)
    {
        this->generasi = generasi;
    }

    void setJumlahCore(int jumlah_core)
    {
        this->jumlah_core = jumlah_core;
    }

    string getNamaSeri()
    {
        return nama_seri;
    }

    int getGenerasi()
    {
        return generasi;
    }

    int getJumlahCore()
    {
        return jumlah_core;
    }

    ~CPU() {}
};

File: python/CPU.py
from Komponen import Komponen

class CPU(Komponen):
    def __init__(self, nama_seri="", generasi=0, jumlah_core=0, nama="", merk="", tahun_rilis=0, harga=0):
        super().__init__(nama, merk, tahun_rilis, harga)
        self.nama_seri = nama_seri
        self.generasi = generasi
        self.jumlah_core = jumlah_core
    
    def setNamaSeri(self, nama_seri):
        self.nama_seri = nama_seri
    
    def setGenerasi(self, generasi):
        self.generasi = generasi
    
    def setJumlahCore(self, jumlah_core):
        self.jumlah_core = jumlah_core
    
    def getNamaSeri(self):
        return self.nama_seri
    
    def getGenerasi(self):
        return self.generasi
    
    def getJumlahCore(self):
        return self.jumlah_core